src/Cake/Program.cs

```diff
@@ -76,6 +76,18 @@ private static IContainer CreateContainer(bool mono)
             builder.RegisterType<WindowsRegistry>().As<IRegistry>().SingleInstance();
             builder.RegisterType<CakeContext>().As<ICakeContext>().SingleInstance();
 
+            // NuGet addins support
+            builder.RegisterType<NuGetVersionUtilityAdapter>()
+                .As<INuGetFrameworkCompatibilityFilter>()
```

Comment:
I've done it both ways in my own work, and I think I agree it's better to use the more explicit syntax.  It just makes it easier to find usages of the interface.